libminifi/src/utils/tls/DistinguishedName.cpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utils/tls/DistinguishedName.h"
+
+#include <algorithm>
+
+#include "utils/StringUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+namespace tls {
+
+DistinguishedName::DistinguishedName(const std::vector<std::string>& components) {
+  std::transform(components.begin(), components.end(), std::back_inserter(components_),
+      [](const std::string& component) { return utils::StringUtils::trim(component); });
+  std::sort(components_.begin(), components_.end());
+}
+
+DistinguishedName DistinguishedName::fromCommaSeparated(const std::string& comma_separated_components) {
+  return DistinguishedName{utils::StringUtils::split(comma_separated_components, ",")};
+}
+
+DistinguishedName DistinguishedName::fromSlashSeparated(const std::string &slash_separated_components) {
+  return DistinguishedName{utils::StringUtils::split(slash_separated_components, "/")};
```

Comment:
This is called zero to 2 or 3 times per heartbeat if `nifi.security.use.system.cert.store=true` is set and `nifi.security.windows.client.cert.cn` is non-empty.  The number of elements in `slash_separated_components` is also small, typically 2 to 4.

I think a better performance improvement would be to create the `SSLContextService` object used in the heartbeat only once at startup, rather than at each heartbeat.  I am planning to do that later in a separate PR.

Reply:
That's a great idea, I'm fine with leaving it as is in the meantime. I think deployments with the most frequent heartbeat intervals are around 1/sec, so it should be fine in either case.